src/main/java/net/neoforged/moddevgradle/internal/ModDevRunWorkflow.java

```diff
@@ -0,0 +1,490 @@
+import java.util.function.Consumer;
+
+/**
+ * After modding has been enabled, this will be attached as an extension to the project.
+ */
+public class ModDevRunWorkflow {
+    private static final String EXTENSION_NAME = "__internal_modDevRunWorkflow";
+
+    /**
+     * This must be relative to the project directory since we can only set this to the same project-relative
+     * directory across all subprojects due to IntelliJ limitations.
+     */
+    static final String JUNIT_GAME_DIR = "build/minecraft-junit";
+
+    private final Project project;
+    private final Branding branding;
+    @Nullable
+    private final ModuleDependency modulePathDependency;
+    @Nullable
+    private final ModuleDependency testFixturesDependency;
+    private final ModuleDependency gameLibrariesDependency;
+    private final Configuration additionalClasspath;
+    private final @Nullable Configuration userDevConfigOnly;
+
+    /**
+     * @param gameLibrariesDependency A module dependency that represents the library dependencies of the game.
+     *                                This module can be depended on with different usage attributes, which allow it
+     *                                to expose different sets of libraries for use in compiling code or at runtime
+     *                                (apiElements vs. runtimeElements).
+     */
+    private ModDevRunWorkflow(Project project,
+                              Branding branding,
+                              ModDevArtifactsWorkflow artifactsWorkflow,
+                              @Nullable ModuleDependency modulePathDependency,
+                              @Nullable ModuleDependency runTypesConfigDependency,
+                              @Nullable ModuleDependency testFixturesDependency,
+                              ModuleDependency gameLibrariesDependency,
+                              DomainObjectCollection<RunModel> runs,
+                              VersionCapabilities versionCapabilities) {
+        this.project = project;
+        this.branding = branding;
+        this.modulePathDependency = modulePathDependency;
+        this.testFixturesDependency = testFixturesDependency;
+        this.gameLibrariesDependency = gameLibrariesDependency;
+
+        var configurations = project.getConfigurations();
+
+        // Let's try to get the userdev JSON out of the universal jar
+        // I don't like having to use a configuration for this...
+        if (runTypesConfigDependency != null) {
+            userDevConfigOnly = configurations.create("neoForgeConfigOnly", spec -> {
+                spec.setDescription("Resolves exclusively the NeoForge userdev JSON for configuring runs");
+                spec.setCanBeResolved(true);
+                spec.setCanBeConsumed(false);
+                spec.setTransitive(false);
+                spec.getDependencies().add(runTypesConfigDependency);
+            });
+        } else {
+            // Create an implicit configuration
```

Comment:
What does this mean? The configuration is null.

Reply:
Removed.